src/ngx_http_lua_shdict.c

```diff
@@ -401,7 +404,13 @@ ngx_http_lua_shdict_get_helper(lua_State *L, int get_stale)
                           "but only seen %d", n);
     }
 
-    zone = lua_touserdata(L, 1);
+    if (LUA_TTABLE != lua_type(L, 1)) {
```

Comment:
Style: nginx always place constants to be compared on the right hand side of the comparison operator. I do understand the rationale behind this style but modern C compilers already emit this kind of typos (using `=` for `==`) with `-Wall`, like gcc.